unikernel.ml

```diff
@@ -1240,6 +1241,122 @@ struct
                                     ~api_meth:false `Internal_server_error reqd
                                     ()))))))
 
+  let process_unikernel_update ~unikernel_name ~job ~build
+      (unikernel_cfg : Vmm_core.Unikernel.config) (user : User_model.user)
+      albatross http_client reqd =
+    Builder_web.send_request http_client
+      ("/job/" ^ job ^ "/build/" ^ build ^ "/main-binary")
+    >>= function
+    | Error (`Msg err) ->
+        Logs.err (fun m ->
+            m
+              "builds.robur.coop: Error while fetching the binary of %s with \
+               error: %s"
+              unikernel_name err);
+        Middleware.http_response reqd ~title:"Error"
+          ~data:
+            (`String
+               ("An error occured while fetching the binary from \
+                 builds.robur.coop with error " ^ err))
+          `Internal_server_error
+    | Ok image -> (
+        match
+          Albatross.manifest_devices_match ~bridges:unikernel_cfg.bridges
+            ~block_devices:unikernel_cfg.block_devices image
+        with
+        | Error (`Msg err) ->
+            Middleware.http_response reqd ~title:"Error"
+              ~data:(`String ("Manifest mismatch: " ^ err))
+              `Bad_request
+        | Ok () -> (
+            let unikernel_config = { unikernel_cfg with image } in
+            Albatross.query albatross ~domain:user.name ~name:unikernel_name
+              (`Unikernel_cmd (`Unikernel_force_create unikernel_config))
+            >>= function
+            | Error msg ->
+                Logs.err (fun m -> m "Error querying albatross: %s" msg);
+                Middleware.http_response reqd ~title:"Error"
+                  ~data:(`String ("Error querying albatross: " ^ msg))
+                  `Internal_server_error
+            | Ok (_hdr, res) -> (
+                match Albatross_json.res res with
+                | Error (`String err) ->
+                    Middleware.http_response reqd ~title:"Error"
+                      ~data:(`String (String.escaped err))
+                      `Internal_server_error
+                | Ok res ->
+                    Logs.info (fun m ->
+                        m "%s has been updated succesfully with result: %s"
+                          unikernel_name
+                          (Yojson.Basic.to_string res));
+                    Middleware.http_response reqd ~title:"Update Successful"
+                      ~data:
+                        (`String
+                           (unikernel_name
+                          ^ " has been updated to the latest build."))
```

Comment:
maybe to build <ID> where ID could be the sha256 or the build UUID so that the operator could look it up (since there may be while updating an even newer build be produced and uploaded to builds.robur.coop).